projects/common/src/time/page-time-range-preference.service.ts

```diff
@@ -22,13 +22,11 @@ export class PageTimeRangePreferenceService {
     this.pageTimeRangeStringDictionary$ = this.buildPageTimeRangeObservable();
   }
 
-  public getTimeRangePreferenceForPage(path: string): Observable<TimeRange> {
+  public getTimeRangePreferenceForPage(path: string): Observable<TimeRange | undefined> {
```

Comment:
I think we'd want this to always return a value right? The only time it wouldn't would be where no default is assigned for a path, so it seems reasonable to default in here rather than in the caller (I'm assuming left nav?) given that the rest of the values are being assigned in here.

Reply:
Yeah, looked back at the code - this is being called by left nav service which assigns the values to each nav entry. In that component, if the nav entry is missing a value, we assign a default - but if this always returns a value, we can keep all the time ranges coming from the same layer.

Reply:
Previously it returne `timeRangeService.getCurrentTimeRange` which was a circular dependency, which is why it was removed from here. 

The 'global' default wasn't added here because If we don't specify the default value in the nav list then then two default values need to be maintained: one for the nav list when the FF is disabled, and one for this file when it is enabled - because the nav items do not get decorated when the FF is disabled, so the logic in `activeItem$` wouldn't have a default.

Reply:
I agree with not defining it in two places, but I think there's plenty of ways to approach avoiding that which doesn't require defining the default global time range in a nav component. For example, we could always decorate and use the feature flag to control whether we read + write preferences only.